Wrappers/Python/cil/processors/MaskGenerator.py

```diff
@@ -0,0 +1,338 @@
+                axis = data.dimension_labels.index(self.axis)
+                ndim = data.number_of_dimensions
+                
+                kernel = [1] * ndim
+                kernel[axis] = self.window
+                kernel = tuple(kernel)
+
+                mean_array = ndimage.generic_filter(arr, numpy.mean, size=kernel, mode='reflect')
+                std_array = ndimage.generic_filter(arr, numpy.std, size=kernel, mode='reflect')
+
+                mask[numpy.abs(arr - mean_array) > self.threshold_factor * std_array] = 0
+            
+            # if global movmean
+            else:
+                
+                ndim = data.number_of_dimensions
+
+                mean_array = ndimage.generic_filter(arr, numpy.mean, size=(self.window,)*ndim, mode='reflect')
+                std_array = ndimage.generic_filter(arr, numpy.std, size=(self.window,)*ndim, mode='reflect')
+                
+                mask[numpy.abs(arr - mean_array) > self.threshold_factor * std_array] = 0          
+            
+        elif self.mode == 'movmedian':
+
+            c = -1 / (numpy.sqrt(2) * special.erfcinv(3 / 2))
+            
+            # if movmedian along specific axis
+            if self.axis is not None:
+                
+                axis = data.dimension_labels.index(self.axis)
+                ndim = data.number_of_dimensions
+                
+                # construct filter kernel
+                kernel_shape = []
+                for i in range(ndim):
+                    if i == axis:
+                        kernel_shape.append(self.window)
+                    else:
+                        kernel_shape.append(1)
+                
+                kernel_shape = tuple(kernel_shape)
+                
+                median_array = ndimage.median_filter(arr, footprint=kernel_shape, mode='reflect')
+                
+                tmp = abs(arr - median_array)
+                mask[tmp > self.threshold_factor * c * ndimage.median_filter(tmp, footprint=kernel_shape, mode='reflect')] = 0
+            
+            # if global movmedian
+            else:
+                
+                # construct filter kernel
+                ndim = data.number_of_dimensions
+                kernel_shape = tuple([self.window]*ndim)
+                median_array = ndimage.median_filter(arr, size=kernel_shape, mode='reflect')
+                
+                tmp = abs(arr - median_array)
+                mask[tmp > self.threshold_factor * c * ndimage.median_filter(tmp, size=kernel_shape, mode='reflect')] = 0
+        
+        else:
+            raise ValueError('Mode not recognised. One of the following is expected: ' + \
+                              'special_values, nan, inf, threshold, quantile, mean, median, movmean, movmedian')
+        
+        data.fill(arr)     
+        return DataContainer(mask, deepcopy=True, dimension_labels=data.dimension_labels)
+        
+
+    def _parse_threshold_value(self, arr):
```

Comment:
`_parse_threshold_value` and `_parse_quantile_value` are fundamentally the same code, with the only difference is the use of `numpy.amax` and `numpy.quantile`?

I think we could think of making this a little more maintainable by using a single function?

Reply:
fixed